### PR TITLE
Add Duplicate & DuplicateBy

### LINKED ISSUE #121: Add Duplicate & DuplicateBy
It would be useful to have:
`func Duplicate[T comparable](collection []T) []T `
`func DuplicateBy[T any, U comparable](collection []T, iteratee func(T) U) []T`

That would only return the duplicated values of the collection
.

### PR DESCRIPTION
**Pull request by CorentinClabaut**
closes #121

### CODE REVIEW

**Comment by CorentinClabaut** [comment]
That's a good point. 
I'm happy to make the change but before I'll wait on @samber input to be sure he is interested in the function being added.

**Review by samber — CHANGES_REQUESTED** [review]
Hi there,

`Duplicates` looks better than `Duplicate` 👍 

But since this function is not exactly the opposite of `Uniq`, I would name it `FindDuplicates` or `OnlyDuplicates`. WDYT ?

Also:
- is it a "search helper" (find.go) or a "slice helper" ?
- what about the opposite: `FindUniq`/`OnlyUniq` ?

Feel free to suggest other names.

File: slice.go [hunk 1]
```diff
@@ -123,6 +123,63 @@ func UniqBy[T any, U comparable](collection []T, iteratee func(T) U) []T {
 	return result
 }
 
+// Duplicate returns a slice with the first occurence of each duplicated elements of the collection.
+// The order of result values is determined by the order they occur in the collection.
+func Duplicate[T comparable](collection []T) []T {
+	isDupl := make(map[T]bool, len(collection))
+
+	for _, item := range collection {
```
[the remark this answers is not in the document]
Comment (CorentinClabaut) [reply slice.go:131]: That's a good point.
I didn't do it in a single loop to copy the first occurence of each duplicated elements.
But as the elements must be comparable I guess it doesn't make any difference.

**Comment by CorentinClabaut** [comment]
Good points!

I'll rename Duplicate/DuplicateBy to FindDuplicates/FindDuplicatesBy and I'll create FindUniques/FindUniquesBy.
I'll also move these functions to find.go

File: slice.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (samber) [reply slice.go]: hell yeah !

i didn't realize order was not conserved

**Pull request merged**